README
#######################################################
# Phynd Network Indexer

# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
#######################################################

##      Phynd Network Indexer Readme Document
##      Written by Brent Holden <brent@phynd.net>
##      Copyright Brent Holden        08-07-2003


The Phynd Network Indexer Project Members are:
========================================================
David Borgeson <david@phynd.net>
Brent Holden <brent@phynd.net>


Description
========================================================

The Phynd Network Indexer is a Windows Networking cataloging agent that queries a list of 
configurable IP addresses to find Windows (Samba) shares and catalogs their contents into a 
searchable MySQL database.  This database can be searched through a world wide web 
interface.


How To Install 
========================================================

The notes on how to install Phynd are located in the INSTALL document. The INSTALL document will 
go over a brief description of how to extract the Phynd tarball and how to get the catalog agent 
up and running.  The INSTALL document does NOT go over how to install CPAN modules, how to 
configure the Apache webserver, or how to configure MySQL.  These software packages are assumed 
to be installed and correctly configured before Phynd is ever installed.


Notes
========================================================

This program does not provide a listing of any hidden shares within your network.  The Phynd 
indexer respects the privacy of those not wishing to make their shares public.  Any publicly 
shared host containing files that is configured to be cataloged will be cataloged, with the 
forementioned exception.

The authors and people involved with this project take no responsibility for any 
irresponsibility, negligence, or misuse on the part of the end user using this software, nor 
the person providing this software for public or private use.  This software is provided with an 
interface to remove violating keywords from a search listing, and is intended to be used by 
sites in correspondence with the DMCA "Safe Harbour" provisions if it is deemed necessary to 
remove any search listings.

Any violations incurred by the end user of this software is of their own will, and their 
responsibility.


License
========================================================

This program is distributed under the GNU GPL2.



CHANGELOG
----------------------
Phynd - Changelog
----------------------

2003-12-21	David Borgeson <koala@phynd.net>
	* Changed the database structure to be more efficient
	* Consolidated the configuration files in /etc/phynd
	* Added mutex locking to the catalog.pl script



2003-08-07 David Borgeson  <david@phynd.net>
	* Initial release of Phynd 5.0


Major differences from Phynd 4.0

	* Multithreaded PERL cataloging agent
	* MySQL catalog rather than flatfile databases
	* PHP Web Interface
	* Web Based Configuration
	* Word Restriction
	* Enhanced search access configuration
	* Enhanced catalog range configuration
	* Plugin architecture removed for future redesign and implementation



config/phyndconfig-ip-post.php
<?
# Phynd Network Indexer

#######################################################
# Phynd Network Indexer

# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License
# as published by the Free Software Foundation; either version 2
# of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
#######################################################

##	Phynd Network Indexer Configuration
##	Written by David Borgeson <borgeson@engr.uconn.edu>
##	Copyright David Borgeson	2003
?>

<html>
<head>
<title>Phynd Configuration</title>
<?


require_once("/etc/phynd/phyndvars.php");


$POSTPAGE = "phyndconfig-ip-post.php";

@$page = $_GET['page'];
@$cat = $_GET['cat'];



switch ($page)
{

case "access":

	switch ($cat)
	{
	case "includeip":
	$TABLE = $ACCESSINCLUDEIP;
	$title = "ACCESS Include IP's";
	break;

	case "excludeip":
	$TABLE = $ACCESSEXCLUDEIP;
	$title = "ACCESS Exclude IP's";
	break;

	case "netblocks":
	$TABLE = $ACCESSNETBLOCKS;
	$title = "ACCESS Netblocks";
	break;

	default:
	exit;
	}

break;


case "catalog":

	switch ($cat)
	{
	case "includeip":
	$TABLE = $CATALOGINCLUDEIP;
	$title = "CATALOG Include IP's";
	break;

	case "excludeip":
	$TABLE = $CATALOGEXCLUDEIP;
	$title = "CATALOG Exclude IP's";
	break;

	case "restrictwords":
	$TABLE = $CATALOGRESTRICTWORDS;
	$title = "CATALOG Restrict Words";
	break;

	case "netblocks":
	$TABLE = $CATALOGNETBLOCKS;
	$title = "CATALOG Netblocks";
	break;


	default:
	exit;
	}

break;

default:
exit;

}


$dbase = mysql_connect($DATABASEHOST,$DATABASEUSER, $DATABASEPASS);		# connect to database
mysql_select_db($DATABASE);							# select the table



if ($_GET['action'] == "add" && ($_GET['cat'] == "includeip" || $_GET['cat'] == "excludeip"))	# want to add to an iplist
	{
	$ip = $_POST['ip'];

	$query ="INSERT INTO `$TABLE` (`ip`) VALUES ('$ip') ";
	mysql_query($query);
	echo mysql_error();
	}


else if ($_GET['action'] == "delete" && ($_GET['cat'] == "includeip" || $_GET['cat'] == "excludeip"))	# want to delete fron an iplist
	{
	mysql_query("DELETE FROM `$TABLE` WHERE `ip` = '$ip'");
	}


else if ($_GET['action'] == "add" && $_GET['cat'] == "netblocks")	# want to add to netblocks
	{
	$ip1 = $_POST['start'];
	$ip2 = $_POST['end'];

	mysql_query("INSERT INTO $TABLE (`start` , `end`) VALUES ('$ip1' , '$ip2') ");


	}
else if ($_GET['action'] == "add" && $_GET['cat'] == "restrictwords")
	{

	$word = $_POST['word'];
	mysql_query("INSERT INTO $TABLE (`word`) VALUES ('$word') ");


	}

else if ($_GET['action'] == "delete" && $_GET['cat'] == "restrictwords")
	{

	$word = $_GET['word'];
	mysql_query("DELETE FROM `$TABLE` WHERE `word` = '$word' ");
	echo mysql_error();


	}







else		# want to delete from netblocks
	{

	$ip1 = $_GET['ip1'];
	$ip2 = $_GET['ip2'];

	mysql_query("DELETE FROM `$TABLE` WHERE `start` = '$ip1' AND `end` = '$ip2' ");
	echo mysql_error();
	}


$backpage = $_SERVER['HTTP_REFERER'];



mysql_close($dbase);			# close the database




echo "<meta http-equiv=\"refresh\" content=\"0; url=$backpage\">";

?>
</head>

<body>

</body>

</html>
